### PR TITLE
Resolve build errors on app-router module

### PR DESCRIPTION
**Pull request by jstr1121**
- Update protobuf for router module
- Connect router module to app.go
- Resolve build issues for router module
- Resolve ParseReceiverData parsing function part

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -117,6 +117,9 @@ import (
 	interchainquerykeeper "github.com/Stride-Labs/stride/v3/x/interchainquery/keeper"
 	interchainquerytypes "github.com/Stride-Labs/stride/v3/x/interchainquery/types"
 
+	router "github.com/Stride-Labs/stride/v3/x/app-router"
```
Comment (asalzmann) [inline app/app.go:120]: This will need to be updated to v4 soon

File: x/app-router/types/expected_keepers.go [hunk 2]
```diff
@@ -3,10 +3,31 @@ package types
...
+	stakeibctypes "github.com/Stride-Labs/stride/v3/x/stakeibc/types"
 )
 
 // TransferKeeper defines the expected transfer keeper
 type StakeibcKeeper interface {
 	LiquidStake(goCtx context.Context, msg *stakeibctypes.MsgLiquidStake) (*stakeibctypes.MsgLiquidStakeResponse, error)
 }
+
+// AccountKeeper defines the contract required for account APIs.
+type AccountKeeper interface {
```
Comment (asalzmann) [inline x/app-router/types/expected_keepers.go:18]: Where are these used?

**Review by asalzmann — APPROVED** [review]
Looks good! Can you add a description to this pull request?

After merging this, good next steps would be
- Add an upgrade handler / set the store key in order to add this module
- Add unittests
- Add integration tests

File: x/app-router/types/expected_keepers.go (hunk 2 above)
Comment (jstr1121) [reply x/app-router/types/expected_keepers.go:18]: Removed on later commits

File: app/app.go (hunk 1 above)
Comment (jstr1121) [reply app/app.go:120]: Okay

**Pull request merged**